CRM/Contribute/BAO/Query.php

```diff
@@ -227,6 +227,17 @@ public static function whereClauseSingle(&$values, &$query) {
         $query->_tables['civicrm_contribution'] = $query->_whereTables['civicrm_contribution'] = 1;
         return;
 
+      case 'contribution_cancel_date':
+      case 'contribution_cancel_date_low':
+      case 'contribution_cancel_date_low_time':
+      case 'contribution_cancel_date_high':
+      case 'contribution_cancel_date_high_time':
+        // process to / from date
+        $query->dateQueryBuilder($values,
+          'civicrm_contribution', 'contribution_cancel_date', 'cancel_date', 'Contribution Cancel Date'
```

Comment:
Can you wrap "Contribution Cancel Date" in `ts()` ? -- and also the 2-3 other occurrences of `dateQueryBuilder()` which should have been wrapped as well.

Reply:
.. and possibly use the same label as "Cancelled / Refunded Date"? (This is shown in the search results, I think?)

Reply:
@mlutfy sure.. what do you think about here (line 1066):

```
    // we only have recurring dates using this ATM so lets' short cut to find the table name
    $table = 'contribution_recur';
    $fieldName = explode($table . '_', $field);
    $query->dateQueryBuilder($values,
      'civicrm_' . $table, $field, $fieldName[1], $title
    );
    return TRUE;
  }
```



should I wrap it as 
```
$query->dateQueryBuilder($values,
      'civicrm_' . $table, $field, $fieldName[1], ts($title)
    );
```
?

Reply:
@sluc23 No, because gettext/civistrings uses the `ts()` function to detect which strings to extract (to send to Transifex). So while `ts($title)` would send the text through gettext, the actual string would not be present in the translation dictionary.
In other words, `ts` means:
- "lookup this string in the translation dictionary (civicrm.mo)"
- "and send this string to Transifex".
Hence a string must not include a variable, since we do not know what the value of the variable is during extraction time (which is done as a c-i process outside program execution).